source/MaterialXCore/Document.cpp

```diff
@@ -169,11 +169,20 @@ void Document::initialize()
 }
 
 NodeDefPtr Document::addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const string& nodeDefName, const string& node,
```

Comment:
This method doesn't seem to be used within the MaterialX codebase (other than in unit tests), and its purpose seems a bit ambiguous to me.  Perhaps this should instead be broken down into multiple simpler methods, rather than adding another option to its argument list?

Reply:
I'll revert this for now and revisit. This change allowed defining a namespace while add nodes. I don't think this commonly used but the Autodesk nodes have namespaces. https://github.com/autodesk-forks/MaterialX/blob/adsk_contrib/dev/libraries/adsk/adsklib/adsklib_defs.mtlx#L106